libnavigation-base/src/main/java/com/mapbox/navigation/base/route/model/RouteLegProgressNavigation.kt

```diff
@@ -0,0 +1,262 @@
+     * is currently on.
+     *
+     * @return a [RouteStepProgressNavigation] object
+     */
+    fun currentStepProgress(): RouteStepProgressNavigation? = currentStepProgress
+
+    /**
+     * Provides a list of points that represent the current step
+     * step geometry.
+     *
+     * @return list of points representing the current step
+     */
+    fun currentStepPoints(): List<Point>? = currentStepPoints
+
+    /**
+     * Provides a list of points that represent the upcoming step
+     * step geometry.
+     *
+     * @return list of points representing the upcoming step
+     */
+    fun upcomingStepPoints(): List<Point>? = upcomingStepPoints
+
+    /**
+     * Not public since developer can access same information from [RouteProgressNavigation].
+     */
+    internal fun routeLeg(): RouteLegNavigation? = routeLeg
+
+    internal fun stepDistanceRemaining(): Double = stepDistanceRemaining
+
+    fun toBuilder() = builder
+
+    class Builder {
+        private var stepIndex: Int = 0
+        private var fractionTraveled: Float = 0f
+        private var distanceTraveled: Double = 0.0
+        private var distanceRemaining: Double = 0.0
+        private var durationRemaining: Double = 0.0
+        private var stepDistanceRemaining: Double = 0.0
+        private var currentStepPoints: List<Point>? = null
+        private var upcomingStepPoints: List<Point>? = null
+        private var previousStep: LegStepNavigation? = null
+        private var upComingStep: LegStepNavigation? = null
+        private var followOnStep: LegStepNavigation? = null
+        private lateinit var _routeLeg: RouteLegNavigation
+        private lateinit var _currentStep: LegStepNavigation
+        private lateinit var currentStepProgress: RouteStepProgressNavigation
+
+        fun stepIndex(stepIndex: Int) = apply { this.stepIndex = stepIndex }
+
+        fun distanceRemaining(distanceRemaining: Double) =
+                apply { this.distanceRemaining = distanceRemaining }
+
+        fun durationRemaining(durationRemaining: Double) =
+                apply { this.durationRemaining = durationRemaining }
+
+        fun currentStep(currentStep: LegStepNavigation) = apply { this._currentStep = currentStep }
+
+        fun currentStepPoints(currentStepPoints: List<Point>?) =
+                apply { this.currentStepPoints = currentStepPoints }
+
+        fun upcomingStepPoints(upcomingStepPoints: List<Point>?) =
+                apply { this.upcomingStepPoints = upcomingStepPoints }
+
+        fun routeLeg(routeLeg: RouteLegNavigation) = apply { this._routeLeg = routeLeg }
+
+        fun stepDistanceRemaining(stepDistanceRemaining: Double) =
+                apply { this.stepDistanceRemaining = stepDistanceRemaining }
+
+        private fun validate() {
+            var missing = ""
+            if (!this::_routeLeg.isInitialized) {
+                missing += " _routeLeg"
```

Comment:
Typo

```suggestion
                missing += " routeLeg"
```

Reply:
I deliberately wrote it that way. If you look at the message it says `RouteLegProgressNavigation.Builder missing required properties: _routeLeg` . Look at the word `.Builder`

Reply:
As a user point of view you don't need to know that it's called `_routeLeg`, the API is `routeLeg` https://github.com/mapbox/mapbox-navigation-android/blob/9f69fba1bed57504e57c6f4f5a3e79c529517eeb/libnavigation-base/src/main/java/com/mapbox/navigation/base/route/model/RouteLegProgressNavigation.kt#L159